tools/guided-fuzzing/include/GuidedExecutionPolicy.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "caffeine/Interpreter/Context.h"
+#include "caffeine/Interpreter/Policy.h"
+#include "caffeine/Solver/Solver.h"
+
+namespace caffeine {
+
+/**
+ * Class to control the behaviour of an Interpreter instance. It also provides a
+ * number of hooks for the policy to execute code when various context-related
+ * events occur within the executor. By default, these do nothing.
+ *
+ * If you don't care about controlling queuing behaviour you may want to derive
+ * from AlwaysAllowExecutionPolicy instead of the base ExecutionPolicy class.
+ */
+class GuidedExecutionPolicy : ExecutionPolicy {
```

Comment:
Needs to be public otherwise this class won't be usable as an `ExecutionPolicy`
```suggestion
class GuidedExecutionPolicy : public ExecutionPolicy {
```